core/src/main/java/de/jplag/comparison/GreedyStringTiling.java

```diff
@@ -26,9 +26,9 @@
 public class GreedyStringTiling {
     private final int minimumMatchLength;
     private final JPlagOptions options;
-    private final Map<Submission, Set<Token>> baseCodeMarkings = new IdentityHashMap<>();
```

Comment:
ConcurrentHashMaps work differnetly from identityHashMaps. I'd suggest to use Collections.synchronizedMap() to ensure that behavior stays the same.

Reply:
We also discussed this PR in the last meeting; the other option would be ConcurrentHashMap + add an equals/hashcode method to Submission. @dfuchss, do you have any opinions architecture-wise? I guess IdentityHashMap will preserve behavior no matter what happens in the future with a potential equals method.

Reply:
Since there is already a equals / hashcode impl. it can be tricky to ensure the same handling as before.

Another option is not to use a map, but to give the submissions some indices. But I think a synchronized map will be fine